services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_service_specs/sidecar.py

```diff
@@ -304,16 +319,26 @@ def get_dynamic_sidecar_spec(
             dynamic_services_scheduler_settings.DYNAMIC_SIDECAR_PROMETHEUS_SERVICE_LABELS,
             scheduler_data.callbacks_mapping,
         )
-        | StandardSimcoreDockerLabels(
-            user_id=scheduler_data.user_id,
-            project_id=scheduler_data.project_id,
-            node_id=scheduler_data.node_uuid,
-            product_name=scheduler_data.product_name,
-            simcore_user_agent=scheduler_data.request_simcore_user_agent,
-            swarm_stack_name=dynamic_services_scheduler_settings.SWARM_STACK_NAME,
-            memory_limit=ByteSize(0),  # this should get overwritten
-            cpu_limit=0,  # this should get overwritten
-        ).to_simcore_runtime_docker_labels(),
+        | standard_simcore_docker_labels
+    )
+
+    # add autoscaling constraints if pricing plan is required
+    if hardware_info and len(hardware_info.aws_ec2_instances) > 0:
+        if len(hardware_info.aws_ec2_instances) == 1:
+            ec2_instance_type: str = hardware_info.aws_ec2_instances[0]
+            service_labels[
+                DOCKER_TASK_EC2_INSTANCE_TYPE_PLACEMENT_CONSTRAINT_KEY
+            ] = ec2_instance_type
+        else:
+            log.warning(
+                "Unexpected number oc aws_ec2_instances %s",
+                hardware_info.aws_ec2_instances,
+            )
```

Comment:
I agree with Pedro, that maybe this warning could even go in the HardwareInfo validator. but nevertheless I think the important thing is that we see when we start having backup machine types. cause most probably this will also need to be adapted, and that the moment I don't know how. so from my point of view you can keep it like this.